File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.java
class Solution {
    public String longestPalindrome(String s) {
        String result = "";
        for (int i = 0; i < s.length(); i++) {
            String odd = this.expand(s, i, i);
            String even = this.expand(s, i, i + 1);

            String palindrome = odd.length() > even.length() ? odd : even;
            if (palindrome.length() > result.length()) {
                result = palindrome;
            }
        }
        return result;
    }

    public String expand(String s, int left, int right) {
        while (
            left >= 0 && 
            right < s.length() &&
            s.charAt(left) == s.charAt(right)
        ) {
            left--;
            right++;
        }
        return s.substring(left + 1, right);
    }
}

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.php
class Solution {

    /**
     * @param String $s
     * @return String
     */
    function longestPalindrome($s) {
        $result = "";
        for ($i = 0; $i < strlen($s); $i++) {
            $odd = $this->expand($s, $i, $i);
            $even = $this->expand($s, $i, $i + 1);
            $palindrome = strlen($odd) > strlen($even) ? $odd : $even;
            if (strlen($palindrome) > strlen($result)) {
                $result = $palindrome;
            } 
        }
        return $result;
    }

    function expand($s, $left, $right) {
        while (
            $left >=0 &&
            $right < strlen($s) &&
            $s[$left] === $s[$right]
        ) {
            $left--;
            $right++;
        }
        return substr($s, $left + 1, $right - 1);
    }
}